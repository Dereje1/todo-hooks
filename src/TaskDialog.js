import React, { useState, useEffect } from 'react';
import DialogTitle from "@mui/material/DialogTitle";
import DialogContent from "@mui/material/DialogContent";
import DialogActions from "@mui/material/DialogActions";
import Dialog from "@mui/material/Dialog";
import Button from '@mui/material/Button';
import TextField from '@mui/material/TextField';
import DialogContentText from '@mui/material/DialogContentText';

const TaskDialog = ({
    open,
    handleCancel,
    handleOk,
}) => {
    const [name, setTaskName] = useState('');
    const [description, setTaskDescription] = useState('');
    return (
        <Dialog
            sx={{ "& .MuiDialog-paper": { width: "80%", maxHeight: 530 } }}
            maxWidth="xs"
            open={open}
        >
            <DialogTitle>Add a Task</DialogTitle>
            <DialogContent>
                <DialogContentText>
                    To add a new TODO, please add a name and a description.
                </DialogContentText>
                <TextField
                    autoFocus
                    margin="dense"
                    id="name"
                    label="Name"
                    type="text"
                    fullWidth
                    variant="standard"
                    value={name}
                    onChange={({ target: { value } }) => setTaskName(value)}
                />
                <TextField
                    autoFocus
                    margin="dense"
                    id="description"
                    label="Description"
                    type="text"
                    fullWidth
                    variant="standard"
                    value={description}
                    onChange={({ target: { value } }) => setTaskDescription(value)}
                />
            </DialogContent>
            <DialogActions>
                <Button autoFocus onClick={handleCancel}>
                    Cancel
                </Button>
                <Button onClick={() => handleOk({ name, description })}>Ok</Button>
            </DialogActions>
        </Dialog>
    )

};

export default TaskDialog;