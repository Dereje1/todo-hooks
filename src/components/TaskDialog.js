import React, { useState } from 'react';
import PropTypes from "prop-types";
/* MUI */
import DialogTitle from "@mui/material/DialogTitle";
import DialogContent from "@mui/material/DialogContent";
import DialogActions from "@mui/material/DialogActions";
import Dialog from "@mui/material/Dialog";
import Button from '@mui/material/Button';
import TextField from '@mui/material/TextField';
import DialogContentText from '@mui/material/DialogContentText';

const TaskDialog = ({
    open,
    handleCancel,
    handleOk,
    editableTask
}) => {
    const [name, setTaskName] = useState(editableTask ? editableTask.name : '');
    const [description, setTaskDescription] = useState(editableTask ? editableTask.description : '');
    return (
        <Dialog
            sx={{ "& .MuiDialog-paper": { width: "80%", maxHeight: 530 } }}
            maxWidth="xs"
            open={open}
        >
            <DialogTitle>{editableTask ? `Edit- ${editableTask.name.slice(0,15)}...` : 'Add a Task'}</DialogTitle>
            <DialogContent>
                <DialogContentText>
                    {
                        editableTask ? '' : 'To add a new TODO, please add a name and a description.'
                    }
                </DialogContentText>
                <TextField
                    autoFocus
                    margin="dense"
                    id="name"
                    label="Name"
                    type="text"
                    fullWidth
                    variant="standard"
                    required
                    value={name}
                    onChange={({ target: { value } }) => setTaskName(value)}
                />
                <TextField
                    autoFocus
                    margin="dense"
                    id="description"
                    label="Description"
                    type="text"
                    fullWidth
                    variant="filled"
                    multiline
                    rows={4}
                    value={description}
                    onChange={({ target: { value } }) => setTaskDescription(value)}
                />
            </DialogContent>
            <DialogActions>
                <Button autoFocus onClick={handleCancel}>
                    Cancel
                </Button>
                <Button onClick={() => handleOk({ name, description }, editableTask)}>Ok</Button>
            </DialogActions>
        </Dialog>
    )

};

export default TaskDialog;

TaskDialog.propTypes = {
    handleCancel: PropTypes.func.isRequired,
    handleOk: PropTypes.func.isRequired,
    open: PropTypes.bool.isRequired,
    editableTask: PropTypes.object
}