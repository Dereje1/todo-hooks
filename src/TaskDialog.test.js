import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import TaskDialog from "./TaskDialog";

const props = {
    open: true,
    handleCancel: jest.fn(),
    handleOk: jest.fn(),
}

test('will render', () => {
    const wrapper = shallow(<TaskDialog {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot();
})

test('will control the name field', () => {
    const wrapper = shallow(<TaskDialog {...props} />)
    let nameField = wrapper.find({id: 'name'})
    expect(nameField.props().value).toBe('')
    nameField.props().onChange({target:{value:'stub-name'}})
    nameField = wrapper.find({id: 'name'})
    expect(nameField.props().value).toBe('stub-name')
})

test('will control the description field', () => {
    const wrapper = shallow(<TaskDialog {...props} />)
    let descriptionField = wrapper.find({id: 'description'})
    expect(descriptionField.props().value).toBe('')
    descriptionField.props().onChange({target:{value:'stub-description'}})
    descriptionField = wrapper.find({id: 'description'})
    expect(descriptionField.props().value).toBe('stub-description')
})

test('will send field values to parent on Ok click', () => {
    const wrapper = shallow(<TaskDialog {...props} />)
    let nameField = wrapper.find({id: 'name'})
    let descriptionField = wrapper.find({id: 'description'})
    nameField.props().onChange({target:{value:'stub-name'}})
    descriptionField.props().onChange({target:{value:'stub-description'}})
    const okButton = wrapper.find('ForwardRef(Button)').at(1)
    okButton.props().onClick();
    expect(props.handleOk).toHaveBeenCalledWith({"description": "stub-description", "name": "stub-name"})
})