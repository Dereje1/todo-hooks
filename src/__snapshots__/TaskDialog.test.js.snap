// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`will render 1`] = `
<ForwardRef(Dialog)
  maxWidth="xs"
  open={true}
  sx={
    Object {
      "& .MuiDialog-paper": Object {
        "maxHeight": 530,
        "width": "80%",
      },
    }
  }
>
  <ForwardRef(DialogTitle)>
    Add a Task
  </ForwardRef(DialogTitle)>
  <ForwardRef(DialogContent)>
    <ForwardRef(DialogContentText)>
      To add a new TODO, please add a name and a description.
    </ForwardRef(DialogContentText)>
    <ForwardRef(TextField)
      autoFocus={true}
      fullWidth={true}
      id="name"
      label="Name"
      margin="dense"
      onChange={[Function]}
      type="text"
      value=""
      variant="standard"
    />
    <ForwardRef(TextField)
      autoFocus={true}
      fullWidth={true}
      id="description"
      label="Description"
      margin="dense"
      onChange={[Function]}
      type="text"
      value=""
      variant="standard"
    />
  </ForwardRef(DialogContent)>
  <ForwardRef(DialogActions)>
    <ForwardRef(Button)
      autoFocus={true}
      onClick={[MockFunction]}
    >
      Cancel
    </ForwardRef(Button)>
    <ForwardRef(Button)
      onClick={[Function]}
    >
      Ok
    </ForwardRef(Button)>
  </ForwardRef(DialogActions)>
</ForwardRef(Dialog)>
`;
